common/src/legacy_memory_region.rs

```diff
@@ -1,10 +1,23 @@
 use bootloader_api::info::{MemoryRegion, MemoryRegionKind};
-use core::mem::MaybeUninit;
+use core::{
+    iter::{empty, Empty},
+    mem::MaybeUninit,
+};
 use x86_64::{
     structures::paging::{FrameAllocator, PhysFrame, Size4KiB},
     PhysAddr,
 };
 
+/// A slice of memory that is used by the bootloader and needs to be reserved
+/// in the kernel
+#[derive(Clone, Copy, Debug)]
+pub struct UsedMemorySlice {
+    /// the physical start of the slice
+    pub start: u64,
+    /// the length of the slice
+    pub len: u64,
+}
```

Comment:
Let's use `start` and `end` instead of `start` and `len`. We never use `len` directly, but only use it to calculate the `end`, so let's just store that instead. We can add a constructor to create a `UsedMemorySlice` from `start` + `len` though.